### ISSUE TITLE
Not splitting lines when word wrapped is on makes jumping difficult to pick the location because text position changes

### ISSUE THREAD

**Issue by excetara2** [ice9js/ace-jump-sublime#41]
If you have a document that is soft word-wrapped like markdown or Latex and you issue a `jump_to_word` or `jump_to_char` it changes the word wrapping so it is really hard to visualize where you want to jump because all the text is now moved to a different location.

I noticed the problem is that it doesn't split up a line in the middle (eg. Table~\ref{tbl:chord_data_sets} will move to the next line unstead of being split across two lines). This then propagates down through the text so makes it really difficult to pick a location. Is there anyway around this besides hard wrapping each paragraph as then it works fine or turning word-wrapping off?

**Comment by excetara2** [comment]
Is there a way to tel it to break long words when word wrapping instead of forcing them to the next line?

Like in word wrap plus the setting:

`"WrapPlus.break_long_words"`

I think this could solve this problem because it is forcing breaking these long words to the next line

**Comment by ice9js** [comment]
AceJump shouldn't affect any word wrapping settings in Sublime Text. So if you have it turned off, it should remain that way. If you have it on, it shouldn't change when you use AceJump.

Since I've been unable to reproduce this issue so far, it would be helpful if you could tell me what build of Sublime Text you are using and share a screen cap of the issue.

As for the wrap plus plugin, you could try just installing it and using it alongside AceJump and see if that solves the issue.

**Comment by excetara2** [comment]
Yeah sorry should have done a screen cap earlier to give an example. I am using build 3107 so a recent build.

This is the before:
![image](https://cloud.githubusercontent.com/assets/741078/15551740/b432d2aa-22fa-11e6-9c44-a49f32420e0f.png)

This is the after with two different jumps but same thing happens no matter the front character selected `Nucleus\` wraps around in the first paragraph as well as the `(NMT)`. Here the misalignment isn't so bad but I have times where it basically makes it unusable because so much of the text ends up changing. You can start to see this at the bottom of the page once it has propagated down it gets worse and worse. I have folds I just realized but this happens with or without folds. Maybe it has to do with some latex settings with word wrap:
![image](https://cloud.githubusercontent.com/assets/741078/15551764/c8667c22-22fa-11e6-8f2e-7055e27b3da9.png)

![image](https://cloud.githubusercontent.com/assets/741078/15551802/f28a65c2-22fa-11e6-96cc-be2d2a3ace49.png)

**Comment by ice9js** [comment]
Build 3114 came out recently, and apparently they did some tweaking to word-wrap behaviour. Would you mind trying to update and see if that fixes your issue ? I have seen some improvements on my side.

**Comment by excetara2** [comment]
No it is because when writing latex I don't use word wrap and if there are things like `\ref{blahblahblah-this-is-a-really-long-ref-with-no-line-breaks}` then it will always wrap to the next line when using the acejump but the regular word-wrap breaks these commands so they word wrap at wherever they occur.

**Comment by excetara2** [comment]
Okay I've realized it is something to do with the latex word wrap behaviour is modified because if I copy the text to a blank buffer and paste it. Then \ref{blahblahblah-this-is-a-really-long-ref-with-no-line-breaks} will be broken unless it is the length of the line. I am guessing I need to incorporate this differed behaviour into acejump for latex buffers. I will try to ask the package maintainers because I've submitted some patches to the LaTex plugin to see if they modified the behaviour.

**Comment by ice9js** [comment]
I'm guessing it has to do with the latex syntax definition. Since there is no syntax definition for the AceJump highlighting, it's all just one big comment, when you write a long, continuous expression, it will assume it's just one big word. Latex syntax would actually be able to tell the words in the expression apart, given some non-word characters like `\` occur.

But that's just my guess and I'm not sure how to approach this yet as the plugin has to work independently of the language/syntax highlighting.

**Comment by excetara2** [comment]
Yeah that would make sense that it allows breaking in certain commands like `\cite` and `\ref` because this isn't the case in a normal buffer without latex syntax. I guess there is no easy way to modify the word wrap behaviour like these syntaxes do as you say?

**Comment by excetara2** [comment]
I found this old posts that seems it has to do with the tMlanguage files but yes can affect other syntaxes like python:

https://forum.sublimetext.com/t/word-wrapping-in-latex-files-st-st2/1403/12
